ingest/src/test/java/com/connexta/ingest/transform/TransformClientTest.java

```diff
@@ -0,0 +1,117 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.anything;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.jsonPath;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.method;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.withUnauthorizedRequest;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.net.URISyntaxException;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.web.client.RestTemplate;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class TransformClientTest {
+
+  @Autowired RestTemplate restTemplate;
+
+  @Autowired TransformClient client;
+
+  private MockRestServiceServer mockServer;
+  private String transformEndpoint;
+  private TransformResponse expectedSuccessResponse;
+  private TransformRequest transformRequest;
+  private TransformResponse expectedFailedResponse;
+
+  @Before
+  public void setUp() throws URISyntaxException {
+    mockServer = MockRestServiceServer.createServer(restTemplate);
+    transformEndpoint = "http://localhost:8000/transform";
+    client.setTransformEndpoint(transformEndpoint);
+
+    // Mock data
+    transformRequest = new TransformRequest();
+    transformRequest.setId("1");
+    transformRequest.setBytes("2");
+    transformRequest.setCallbackUrl("callback-url");
+    transformRequest.setMimeType(MediaType.IMAGE_JPEG_VALUE);
+    transformRequest.setProductLocation("product-location");
+    transformRequest.setStagedLocation("staged-location");
+    expectedSuccessResponse = new TransformResponse();
+    expectedSuccessResponse.setId("1");
+    expectedSuccessResponse.setMessage("success");
+    expectedFailedResponse = new TransformResponse();
+    expectedFailedResponse.setId("1");
+    expectedFailedResponse.setMessage("failure");
+    expectedFailedResponse.setDetails(List.of("A", "B"));
+  }
+
+  @Test
+  public void testSuccessfulRequest() throws JsonProcessingException {
+
+    mockServer
+        .expect(requestTo(transformEndpoint))
+        .andExpect(method(HttpMethod.POST))
+        .andExpect(jsonPath("$.id").value("1"))
```

Comment:
✏️ `.andExpect(jsonPath("$.id").value(transformRequest.getId())`
. . .

Reply:
In unit tests I usually test for the actual values. I've had too many cases where the value was not set on the object and that hit an error in the code.